lib.rs

```diff
@@ -478,6 +482,12 @@ impl<A: Array> SmallVec<A> {
 }
 
 impl<A: Array> SmallVec<A> where A::Item: Copy {
+    pub fn from_slice(slice: &[A::Item]) -> Self {
```

Comment:
Also need to make sure A::Item: Copy which is what made me make it more specific (from_slice() rather than impl From).

I'd imagine that a generic From implementation should work regardless of whether it's Copy or not. Without specialization, it's hard to make an efficient version for the Copy type, so I went with a different "from_slice" function.